### ISSUE TITLE
Compilation fails with intel 2024.2 / imp 2021.13

### ISSUE THREAD

**Issue by bourdin** [Parallel-NetCDF/PnetCDF#151]
Trying to build with `./configure --with-mpi=/opt/intel/oneapi/mpi/latest/`. configure fails with the following error message:
```
configure: error: in `/opt/HPC/petsc/main/arch-intel-g/externalpackages/git.pnetcdf':
configure: error: linking to Fortran libraries from C fails
See `config.log' for more details
```
Looking at config.log, I see the following:
```
configure:28226: /opt/intel/oneapi/mpi/latest//bin/mpicc -o conftest -O2   conftest.c   -loopopt=1 -L/opt/intel/oneapi/tbb/2021.13/env/../lib/intel64/g
cc4.8 -L/opt/intel/oneapi/mpi/2021.13/lib -L/opt/intel/oneapi/mkl/2024.2/lib/ -L/opt/intel/oneapi/ippcp/2021.12/lib32/ -L/opt/intel/oneapi/ipp/2021.12/
lib -L/opt/intel/oneapi/dpl/2022.6/lib -L/opt/intel/oneapi/dnnl/2024.2/lib -L/opt/intel/oneapi/dal/2024.6/lib -L/opt/intel/oneapi/compiler/2024.2/lib -
L/opt/intel/oneapi/ccl/2021.13/lib/ -L/opt/intel/oneapi/compiler/2024.2/lib/clang/19/lib/x86_64-unknown-linux-gnu -L/usr/lib/gcc/x86_64-redhat-linux/11
/ -L/usr/lib/gcc/x86_64-redhat-linux/11/../../../../lib64 -L/usr/lib/gcc/x86_64-redhat-linux/11/../../../../lib64/ -L/lib/../lib64 -L/lib/../lib64/ -L/
usr/lib/../lib64 -L/usr/lib/../lib64/ -L/opt/intel/oneapi/tbb/2021.13/env/../lib/intel64/gcc4.8/ -L/opt/intel/oneapi/mpi/2021.13/lib/ -L/opt/intel/onea
pi/ipp/2021.12/lib/ -L/opt/intel/oneapi/dpl/2022.6/lib/ -L/opt/intel/oneapi/dnnl/2024.2/lib/ -L/opt/intel/oneapi/dal/2024.6/lib/ -L/opt/intel/oneapi/co
mpiler/2024.2/lib/ -L/usr/lib/gcc/x86_64-redhat-linux/11/../../../ -L/lib64 -L/lib/ -L/usr/lib64 -L/usr/lib -lifport -lifcoremt -limf -lsvml -lm -lipgo
 -lirc -lpthread -lirc_s -ldl >&5
/usr/bin/ld: cannot find -loopopt=1
icx: error: linker command failed with exit code 1 (use -v to see invocation)
configure:28226: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "PnetCDF"
| #define PACKAGE_TARNAME "pnetcdf"
| #define PACKAGE_VERSION "1.12.3"
| #define PACKAGE_STRING "PnetCDF 1.12.3"
| #define PACKAGE_BUGREPORT "parallel-netcdf@mcs.anl.gov"
| #define PACKAGE_URL "https://parallel-netcdf.github.io"
| #define PACKAGE "pnetcdf"
| #define VERSION "1.12.3"
| #define STDC_HEADERS 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_DLFCN_H 1
| #define LT_OBJDIR ".libs/"
| #define ENABLE_CXX 1
| #define ENABLE_FORTRAN 1
| /* end confdefs.h.  */
| #define FC_DUMMY_MAIN _main
| #ifdef FC_DUMMY_MAIN
| #ifndef FC_DUMMY_MAIN_EQ_F77
| #  ifdef __cplusplus
|      extern "C"
| #  endif
|    int FC_DUMMY_MAIN() { return 1; }
| #endif
| #endif
| int
| main ()
| {
| 
|   ;
|   return 0;
| }
configure:28242: result: unknown
configure:28259: error: in `/opt/HPC/petsc/main/arch-intel-g/externalpackages/git.pnetcdf':
configure:28261: error: linking to Fortran libraries from C fails
```

I can't figure out where does the `-loopopt=1` option comes from. Any idea?

[config.log](https://github.com/user-attachments/files/16968416/config.log)

**Comment by wkliao** [comment]
This issue of `-loopopt=1` has been resolved in the latest release of 1.13.0.
Please give it a try to see if it works for you

**Comment by bourdin** [comment]
Yes it worked. Dunno why I did not try to update to the most version first. I'll boost the version number in PETSc
